code/modules/vending/_vending.dm

```diff
@@ -853,23 +856,34 @@ GLOBAL_LIST_EMPTY(vending_products)
 				if(O.name == N)
 					S = O
 					break
-			if(!account.has_money(S.custom_price))
-				say("You do not possess the funds to purchase this.")
-			else
-				account.adjust_money(-S.custom_price)
-				var/datum/bank_account/owner = private_a
-				if(owner)
-					owner.adjust_money(S.custom_price)
-				use_power(5)
-				vending_machine_input[N] = max(vending_machine_input[N] - 1, 0)
-				if(last_shopper != usr || purchase_message_cooldown < world.time)
-					say("Thank you for buying local and purchasing [S]!")
-					purchase_message_cooldown = world.time + 5 SECONDS
-					last_shopper = usr
-				S.forceMove(drop_location())
-				loaded_items--
-				vend_ready = 1
-	updateUsrDialog()
```

Comment:
You removed dialog update.

Reply:
it updates when it vends the item and it doesnt when you dont have money

Reply:
Alright, thought there were more topic branches above this.